clients/src/main/java/org/apache/kafka/clients/consumer/internals/AsyncKafkaConsumer.java

```diff
@@ -820,7 +823,10 @@ public void seek(TopicPartition partition, OffsetAndMetadata offsetAndMetadata)
                 offsetAndMetadata.leaderEpoch(),
                 currentLeaderAndEpoch);
             updateLastSeenEpochIfNewer(partition, offsetAndMetadata);
-            subscriptions.seekUnvalidated(partition, newPosition);
+
+            Timer timer = time.timer(defaultApiTimeoutMs);
+            SeekUnvalidatedEvent seekUnvalidatedEventEvent = new SeekUnvalidatedEvent(calculateDeadlineMs(timer), partition, newPosition);
+            applicationEventHandler.addAndGet(seekUnvalidatedEventEvent);
```

Comment:
Per the other suggestions, we could make this a bit more succinct:

```suggestion
            SeekUnvalidatedEvent event = new SeekUnvalidatedEvent(
                calculateDeadlineMs(timer),
                partition,
                offsetAndMetadata.offset(),
                offsetAndMetadata.leaderEpoch()
            );
            applicationEventHandler.addAndGet(event);
```